tmb-user-mgt/tmb-userdata-mock/src/main/java/de/interseroh/tmb/user/client/UserInformationServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package de.interseroh.tmb.user.client;
+
+import com.google.gwt.core.client.Callback;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.Cookies;
+import org.fusesource.restygwt.client.MethodCallback;
+import org.gwtbootstrap3.client.ui.AnchorButton;
+import org.gwtbootstrap3.client.ui.base.ComplexWidget;
+import org.gwtbootstrap3.client.ui.constants.ButtonType;
+
+import java.util.logging.Logger;
+
+
+/**
+ * Mock User Service
+ */
+public class UserInformationServiceImpl implements UserInformationService{
+
+    private static final Logger logger = Logger
+            .getLogger(UserInformationServiceImpl.class.getName());
+
+    public UserInformationServiceImpl(String gatewayLocation, String userInfoUrl, String cookiePath, String logoutUrl) {
+        logger.info("USER INFORMATION MOCK HAS BEEN CONFIGURED WITH "
+                +(gatewayLocation==null || gatewayLocation.trim().isEmpty()? " NO GATEWAY ":gatewayLocation)
+                + "\n AND "
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "\n AND " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)